### PR TITLE
added dependency injection objects creation

### PR DESCRIPTION
**Pull request by raj8789**
Dependency injection is done for objects creation of TwitterController class,TwitterRunner class, TwitterImpl class by using @component ,@Autowired  annotation

### CODE REVIEW

File: src/main/java/com/model/SendResponse.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package com.model;
+
+public class SendResponse {
+    private  int statusCode;
+    private String message;
+    private int code;
```
Comment (anasuya1117) [inline src/main/java/com/model/SendResponse.java]: Do a self review before team review

File: src/main/java/com/resource/TwitterController.java [hunk 2]
```diff
@@ -1,63 +1,76 @@
...
-    @POST
-    @Path("/postTweet")
-    @Consumes(MediaType.APPLICATION_JSON)
-    public Response sendTweet(TweetPostRequest tweetPostRequest) {
+    @RequestMapping(method = RequestMethod.POST,value = "postTweet")
+    public ResponseEntity<SendResponse> sendTweet(@RequestBody TweetPostRequest tweetPostRequest) {
         String tweet = tweetPostRequest.getMessage();
-        if (StringUtil.isEmpty(tweet)) {
+        HttpHeaders headers=null;
+        headers = new HttpHeaders();
```
Comment (anasuya1117) [inline src/main/java/com/resource/TwitterController.java]: these two lines can be merged

File: src/main/java/com/resource/TwitterController.java [hunk 3]
```diff
@@ -1,63 +1,76 @@
...
+        headers.add("Custom-Header", "foo");
+        if (StringUtil.isEmpty(tweet))
+        {
             logger.error("Enter a Valid Tweet");
-            return Response.status(400, "Please Enter a valid tweet").build();
-        } else {
-            try {
+            return new ResponseEntity<SendResponse>(
+                    new SendResponse(HttpStatus.BAD_REQUEST_400,"Please enter a Valid Tweet",400),headers,HttpStatus.OK_200);
```
Comment (anasuya1117) [inline src/main/java/com/resource/TwitterController.java]: status code looks conflicting, update

File: src/main/java/com/resource/TwitterController.java [hunk 4]
```diff
@@ -1,63 +1,76 @@
...
+                {
                     logger.error("Tweet Was Not Done Invalid Request");
-                    return Response.status(500, "Request was not completed").build();
+                    return new ResponseEntity<SendResponse>(
+                            new SendResponse(HttpStatus.INTERNAL_SERVER_ERROR_500,"Request tweet is not correct",500),headers,HttpStatus.INTERNAL_SERVER_ERROR_500);
                 }
             } catch (BadRequestException e) {
                 logger.error("Tweet Was Not Done Invalid Request", e);
-                return Response.status(400, "Request tweet is not correct").build();
+                return new ResponseEntity<SendResponse>(
```
Comment (anasuya1117) [inline src/main/java/com/resource/TwitterController.java]: return ResponseStatusException from catch blocks

**Review by anasuya1117 — APPROVED** [review]
Address the comments and merge

File: src/main/java/com/model/SendResponse.java (hunk 1 above)
Comment (raj8789) [reply src/main/java/com/model/SendResponse.java]: removed statuscode, code

File: src/main/java/com/resource/TwitterController.java (hunk 2 above)
Comment (raj8789) [reply src/main/java/com/resource/TwitterController.java]: lines are merged

File: src/main/java/com/resource/TwitterController.java (hunk 3 above)
Comment (raj8789) [reply src/main/java/com/resource/TwitterController.java]: resolve conflicts of status code

File: src/main/java/com/resource/TwitterController.java (hunk 4 above)
Comment (raj8789) [reply src/main/java/com/resource/TwitterController.java]: now returning ResponseStatusException  from catch block

**Pull request merged**